File: app/src/test/java/io/github/tobyhs/weatherweight/TestApp.java
package io.github.tobyhs.weatherweight;

import dagger.android.AndroidInjector;

/**
 * Application class for unit tests
 */
public class TestApp extends App {
    @Override
    protected AndroidInjector<App> applicationInjector() {
        return DaggerTestAppComponent.factory().create(this);
    }
}

File: app/src/test/java/io/github/tobyhs/weatherweight/TestApp.kt
package io.github.tobyhs.weatherweight

import dagger.android.AndroidInjector

/**
 * Application class for unit tests
 */
class TestApp : App() {
    override fun applicationInjector(): AndroidInjector<App> {
        return DaggerTestAppComponent.factory().create(this)
    }
}
